VAMobile/src/screens/HealthScreen/Appointments/PastAppointments/PastAppointments.tsx

```diff
@@ -169,15 +169,11 @@ const PastAppointments: FC<PastAppointmentsProps> = () => {
   }
 
   const getAppointmentsInSelectedRange = (curSelectedRange: PastAppointmentsDatePickerOption, selectedPage: number): void => {
-    dispatch(
```

Comment:
now had to handle a null instance

Reply:
It doesn't look like the date range filter is working for past appointments in this branch

Reply:
here is the log for develop for the date ranges

[0]  DEBUG  logging analytics event vama_segcontrol_click {"label": "Past"}
[0]  LOG  2023-10-12T00:00:00.000-05:00
[0]  LOG  2024-01-11T00:00:00.000-06:00
[0]  DEBUG  logging analytics event vama_modalpick_open {"modal": "Select a date range"}
[0]  DEBUG  logging analytics event vama_modalpick_sel {"modal": "Select a date range", "selection": "5 months to 3 months"}
[0]  LOG  2023-08-01T00:00:00.000-05:00
[0]  LOG  2023-10-12T00:00:00.000-05:00
[0]  DEBUG  logging analytics event vama_modalpick_open {"modal": "Select a date range"}
[0]  DEBUG  logging analytics event vama_modalpick_sel {"modal": "Select a date range", "selection": "8 months to 6 months"}
[0]  LOG  2023-05-01T00:00:00.000-05:00
 LOG  2023-07-31T00:00:00.000-05:00
[0]  DEBUG  logging analytics event vama_modalpick_open {"modal": "Select a date range"}
[0]  DEBUG  logging analytics event vama_modalpick_sel {"modal": "Select a date range", "selection": "11 months to 9 months"}
[0]  LOG  2023-02-01T00:00:00.000-06:00
 LOG  2023-04-30T00:00:00.000-05:00
[0]  DEBUG  logging analytics event vama_modalpick_open {"modal": "Select a date range"}
[0]  DEBUG  logging analytics event vama_modalpick_sel {"modal": "Select a date range", "selection": "All of 2024"}
[0]  LOG  2024-01-01T00:00:00.000-06:00
[0]  LOG  2024-01-11T00:00:00.000-06:00
[0]  DEBUG  logging analytics event vama_modalpick_open {"modal": "Select a date range"}
[0]  DEBUG  logging analytics event vama_modalpick_sel {"modal": "Select a date range", "selection": "All of 2023"}
[0]  LOG  2023-01-01T00:00:00.000-06:00
[0]  LOG  2023-12-31T00:00:00.000-06:00

Reply:
here is this branches log for using the date picker:

[0]  DEBUG  logging analytics event vama_modalpick_sel {"modal": "Select a date range", "selection": "Past 3 months"}
[0]  LOG  2023-10-12T00:00:00.000-05:00
[0]  LOG  2024-01-11T23:59:59.999-06:00
[0]  DEBUG  logging analytics event vama_modalpick_open {"modal": "Select a date range"}
[0]  DEBUG  logging analytics event vama_modalpick_sel {"modal": "Select a date range", "selection": "5 months to 3 months"}
[0]  LOG  2023-08-01T00:00:00.000-05:00
[0]  LOG  2023-10-12T23:59:59.999-05:00
[0]  DEBUG  logging analytics event vama_modalpick_open {"modal": "Select a date range"}
[0]  DEBUG  logging analytics event vama_modalpick_sel {"modal": "Select a date range", "selection": "8 months to 6 months"}
[0]  LOG  2023-05-01T00:00:00.000-05:00
[0]  LOG  2023-07-31T23:59:59.999-05:00
[0]  DEBUG  logging analytics event vama_modalpick_open {"modal": "Select a date range"}
[0]  DEBUG  logging analytics event vama_modalpick_sel {"modal": "Select a date range", "selection": "11 months to 9 months"}
[0]  LOG  2023-02-01T00:00:00.000-06:00
[0]  LOG  2023-04-30T23:59:59.999-05:00
[0]  DEBUG  logging analytics event vama_modalpick_open {"modal": "Select a date range"}
[0]  DEBUG  logging analytics event vama_modalpick_sel {"modal": "Select a date range", "selection": "All of 2024"}
[0]  LOG  2024-01-01T00:00:00.000-06:00
[0]  LOG  2024-01-11T23:59:59.999-06:00
[0]  DEBUG  logging analytics event vama_modalpick_open {"modal": "Select a date range"}
[0]  DEBUG  logging analytics event vama_modalpick_sel {"modal": "Select a date range", "selection": "All of 2023"}
[0]  LOG  2023-01-01T00:00:00.000-06:00
 LOG  2023-12-31T23:59:59.999-06:00

Reply:
Makes sense, we can update the demo mode data to use dynamic dates when we get a chance